src/App.js

```diff
@@ -1,16 +1,133 @@
 import React, { Component } from 'react'
-import logo from './logo.svg'
 import './App.css'
+import PlayMenu from './PlayMenu'
+import Naming from './Naming'
 
 class App extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      game: {
+        gameState: 'naming',
+        difficulty: 1
+      },
+      progress: {
+        days: 12,
+        miles: 123
+      },
+      inventory: {
+        waterFilter: 12,
+        sleepingBags: 3,
+        food: 200
+      },
+      people: [{
+        name: 'You',
+        health: 100,
+        status: 'alive'
+      }, {
+        name: '',
+        health: 100,
+        status: 'dead'
+      }, {
+        name: '',
+        health: 100,
+        status: 'alive'
+      },
+      {
+        name: '',
+        health: 100,
+        status: 'alive'
+      },
+      {
+        name: '',
+        health: 100,
+        status: 'alive'
+      }
+      ]}
+    this.onUserPlay = this.onUserPlay.bind(this)
+    this.handleName = this.handleName.bind(this)
+    this.onConfirm = this.onConfirm.bind(this)
+  }
+
+  rangeGenerator (lowest, highest) {
+    const min = Math.ceil(lowest)
+    const max = Math.floor(highest)
+    return Math.floor(Math.random() * (max - min)) + min
+  }
+
+  // This is causing an error message saying that I am directly mutating state.  Why?
+  walk () {
+    const milesGained = this.rangeGenerator(12, 25)
+    let newMiles = this.state.progress.miles
+    newMiles += milesGained
+    let newDays = this.state.progress.days
+    newDays += 1
+    const peopleLiving = this.state.people.length - this.state.people.filter(function (character) { return character.status === 'dead' }).length
+    const foodLost = this.rangeGenerator(2 * peopleLiving, 5 * peopleLiving)
+    let newFood = this.state.inventory.food
+    newFood -= foodLost
+    this.setState({progress: {miles: newMiles, days: newDays}})
+    this.setState({inventory: {food: newFood}})
+    const lostHealth = this.state.inventory.food > 0 ? 5 : 20
+    const newPeopleList = this.state.people.map(function (character) { character.health -= lostHealth; return character })
+    this.setState({people: newPeopleList})
+  }
+
+  onUserPlay (e) {
+    if (e.keyCode === 13 || e.keyCode === 32) {
```

Comment:
these magic numbers (e.g. `13` and `32`) are relatively meaningless to someone looking at the code. It would be much better to define them as constants. E.g.:
```
const RETURN = 13
const SPACE = 32
...
if (e.keyCode === RETURN || e.keyCode === SPACE) {
```